src/Diagnostics.DataProviders/AscClient.cs

```diff
@@ -0,0 +1,378 @@
+            return client;
+        }
+       );
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AscClient"/> class.
+        /// <param name="config">Config for Asc Data Provider.</param>
+        /// <param name="requestId">AppLens Request Id, used for logging.</param>
+        /// </summary>
+        public AscClient(AscDataProviderConfiguration config, string requestId)
+        {
+            baseUri = config.BaseUri;
+            apiUri = config.ApiUri;
+            apiVersion = config.ApiVersion;
+            userAgent = config.UserAgent;
+        }
+
+        private HttpClient httpClient
+        {
+            get
+            {
+                return client.Value;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<T> MakeHttpPostRequest<T>(string jsonPostBody, string apiUri, string apiVersion, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            string authToken = await AscTokenService.Instance.GetAuthorizationTokenAsync();
+            try
+            {
+                if (string.IsNullOrEmpty(apiUri))
+                {
+                    apiUri = this.apiUri;
+                }
+
+                if (string.IsNullOrEmpty(apiVersion))
+                {
+                    apiVersion = this.apiVersion;
+                }
+
+                string requestUri = baseUri + apiUri + "?api-version=" + apiVersion.Replace("?", string.Empty).Replace("api-version=", string.Empty);
+
+                httpClient.DefaultRequestHeaders.UserAgent.Clear();
+                httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(new ProductHeaderValue(userAgent)));
+
+                using (HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Post, requestUri))
+                {
+                    requestMessage.Headers.Add("Authorization", authToken);
+                    httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", authToken);
+                    requestMessage.Content = new StringContent(jsonPostBody, Encoding.UTF8, "application/json");
+
+                    var response = await httpClient.SendAsync(requestMessage, cancellationToken);
+                    string responseContent = await response.Content.ReadAsStringAsync();
+                    if (response.IsSuccessStatusCode)
+                    {
+                        if (typeof(T).Equals(typeof(string)))
+                        {
+                            return CastTo<T>(responseContent);
+                        }
+                        else
+                        {
+                            T value;
+                            try
+                            {
+                                value = JsonConvert.DeserializeObject<T>(responseContent);
+                            }
+                            catch (Exception serializeException)
```

Comment:
Catch `JsonSerializationException` specifically

Reply:
Thanks for catching this.